recipe/patches/0007-Remove-bazel-version-check.patch

```diff
@@ -0,0 +1,35 @@
+From 0da8732a8d99c436c574bdbb24fe2e35df5e0a7c Mon Sep 17 00:00:00 2001
+From: Jiajun Yao <jjyao@anyscale.com>
+Date: Thu, 9 Dec 2021 08:45:59 -0800
+Subject: [PATCH 7/7] Remove bazel version check
+
+Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>
+---
+ python/setup.py | 11 -----------
+ 1 file changed, 11 deletions(-)
+
+diff --git a/python/setup.py b/python/setup.py
+index 565237825..80613e107 100644
+--- a/python/setup.py
++++ b/python/setup.py
+@@ -424,17 +424,6 @@ def build(build_python, build_java, build_cpp):
+             ] + pip_packages,
+             env=dict(os.environ, CC="gcc"))
+ 
+-    version_info = bazel_invoke(subprocess.check_output, ["--version"])
+-    bazel_version_str = version_info.rstrip().decode("utf-8").split(" ", 1)[1]
+-    bazel_version_split = bazel_version_str.split(".")
+-    bazel_version_digits = [
+-        "".join(takewhile(str.isdigit, s)) for s in bazel_version_split
+-    ]
+-    bazel_version = tuple(map(int, bazel_version_digits))
+-    if bazel_version < SUPPORTED_BAZEL:
+-        logger.warning("Expected Bazel version {} but found {}".format(
+-            ".".join(map(str, SUPPORTED_BAZEL)), bazel_version_str))
```

Comment:
I suggest you add the removal of `SUPPORTED_BAZEL` variable in this patch, too. It would make it so next time upstream changes Bazel version we'll have a merge conflict here which would prompt us to update the Bazel version in the recipe.

Reply:
SUPPORTED_BAZEL is still needed in the setup.py. Also I'm removing that bazel version check code in the upstream as well: https://github.com/ray-project/ray/pull/20990

Reply:
My idea with removing the declaration was to create a conflict _here_ in conda-forge when required Bazel version changes, so `meta.yaml` would be updated accordingly (kind of semi-automatic version check).

Reply:
what about purposely creating a conflict when Bazel required version changes? so `meta.yaml` could be updated when this happens

Reply:
I'd like to defer it to a separate PR: we need a solution not just for bazel but for all dependencies in meta.yaml. Currently I need to manually update them and make sure they are consistent with the versions in the upstream. We need a better solution here.

Reply:
Consistency is semi-automatic with current solution - a merge conflict happens when something is changed in `setup.py` which touches the dependencies. Ideally this should be fully automatic, I totally agree, but I don't have better solution yet.

It would be awesome if you could come up with anything that's better, I would gladly copy it to other conda packages I maintain 🙃

Reply:
I'm not so worried about patching this out. If the recipe builds and passes the tests, then - ipso facto - the bazel version wasn't doing too badly. OTOH, if we're using a bazel version that's really not supported, things will fail loudly anyway.

That said, I don't think it's a bad idea what @vnlitvinov is suggesting - explicit is better than implicit, and as such we'd get a clear reminder (through a rebase conflict) if the supported bazel version changed. But it doesn't have to be done for this PR, IMO, let's just try to keep moving through the queue first.

Thanks for pushing on this BTW, @jjyao! :)